CMakeLists.txt

```diff
@@ -27,92 +25,40 @@ if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU" OR
     add_compile_options(-Wall -Wno-unused-variable)
 endif()
 
-
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 include(AddSYCLExecutable)
+include(AddCTSOption)
 
-# ------------------
-# Extensive mode for full coverage
-option(SYCL_CTS_ENABLE_FULL_CONFORMANCE
-       "Enable full conformance mode with extensive tests" OFF)
-if(SYCL_CTS_ENABLE_FULL_CONFORMANCE)
-    message(STATUS "Full conformance mode: ON")
-    add_host_and_device_compiler_definitions(-DSYCL_CTS_ENABLE_FULL_CONFORMANCE)
-else()
-    message(STATUS "Full conformance mode: OFF")
-    message(WARNING
-            "Full conformance mode should be used for conformance submission")
-endif()
-# ------------------
+add_cts_option(SYCL_CTS_ENABLE_FULL_CONFORMANCE
+    "Enable full conformance with extensive tests" OFF
+    WARN_IF_OFF "Full conformance mode (SYCL_CTS_ENABLE_FULL_CONFORMANCE) should be used for conformance submission")
```

Comment:
In my day-to-day work, I don't run CTS in full mode, so I'll always see a warning. GitHub actions job logs also will have such warning in each log, which doesn't seem useful.

I think it would be nice a have an option to suppress this such warnings or just do not emit it.

Reply:
I personally don't really mind the warning, and I think with this change the situation is already improved as it is printed at the very end, below the configuration summary, making it easier to ignore as it doesn't just whiz by. However I do also question the usefulness of the warning, since to generate the report one would use the script (which sets the flag automatically) anyway. I have some improvements regarding conformance report generation planned, I suggest we revisit this then.